pkg/reconciler/taskrun/resources/taskspec_test.go

```diff
@@ -286,11 +306,224 @@ func TestGetTaskData_VerificationResult(t *testing.T) {
 			Spec:       *sourceSpec.DeepCopy(),
 		}, nil, verificationResult, nil
 	}
-	r, _, err := resources.GetTaskData(context.Background(), tr, getTask)
+	r, _, err := resources.GetTaskData(context.Background(), tr, getTask, getStepAction)
 	if err != nil {
 		t.Fatalf("Did not expect error but got: %s", err)
 	}
 	if d := cmp.Diff(verificationResult, r.VerificationResult, cmpopts.EquateErrors()); d != "" {
 		t.Errorf(diff.PrintWantGot(d))
 	}
 }
+
+func TestGetStepAction(t *testing.T) {
+	tests := []struct {
+		name           string
+		tr             *v1.TaskRun
+		want           *v1.TaskSpec
+		stepActionFunc func(ctx context.Context, n string) (*v1alpha1.StepAction, *v1.RefSource, error)
+	}{{
+		name: "step-action-with-command",
```

Comment:
do we miss args? I can only see command, script, env. And how about having them all in  1 case?

Reply:
We did not. That name was supposed to be "step-action-with-command-and-args". I can also unify them in one case.

Reply:
Actually I cannot since I cannot use command and script together.  so best to leave them separate.

Reply:
Oh yes, maybe 1 case for command, another one for script. and both have arge, env?